CedarJava/src/main/java/com/cedarpolicy/model/slice/Policy.java

```diff
@@ -41,7 +54,8 @@ public Policy(
             throw new NullPointerException("Failed to construct policy from null string");
         }
         if (policyID == null) {
-            throw new NullPointerException("Failed to construct policy with null ID");
+            policyID = "policy" + idCounter;
+            idCounter += 1;
```

Comment:
I think this could cause duplicate IDs if we construct `Policy` objects from multiple threads. I guess we could:
  * document this and leave it as-is
  * use an `AtomicInteger` for the counter (presumably the overhead here isn't massive)
  * allow `null` policy IDs in this class – I think the Rust library's constructors now take `Option`s for policy IDs